hardware/mock/mock.go

```diff
@@ -26,29 +27,26 @@ func (hg HardwareClient) IsHealthy(context.Context) bool {
 // Given any other IP inside the get request, ByIP will return an empty piece of hardware regardless of whether or not the IP
 // actually matches the IP inside `Data`.
 func (hg HardwareClient) ByIP(_ context.Context, ip string) (hardware.Hardware, error) {
-	var hw hardware.Hardware
-	dataModelVersion := os.Getenv("DATA_MODEL_VERSION")
-	switch dataModelVersion {
-	case "1":
-		hw = &hardware.Tinkerbell{}
-
+	switch hg.Model {
+	case datamodel.TinkServer:
+		fmt.Printf("HELLO WORLD\n")
 		if ip != UserIP {
-			return hw, errors.New("rpc error: code = Unknown desc = unexpected end of JSON input")
+			return nil, fmt.Errorf("received non-mock remote ip address: %v", ip)
```

Comment:
should be `errors.Errof` (this package should have been using "pkg/errors" not "errors" like the rest of the repo).

Reply:
Do we really feel the need for `pkg/errors`? Its largely redundant since `fmt.Errorf` was introduced unless we feel stack traces are super important (spoiler: I don't). I've always found well written errors with appropriate context suffice.

Reply:
If nothing else its important to keep in the style with the rest of the repo, but I do think stacks are nice and use them all the time to hone in while debugging production or even tests.

Reply:
I took the liberty of replacing all occurrences of stdlib error and `fmt.Errorf`.

Reply:
Comically, `pkg/errors` doesn't support `%w` necessitating https://github.com/tinkerbell/lint-install/pull/41 😆

Reply:
Youch, good catch!